### PR TITLE
Added VideoCard Menu , and History Page functionalities

### PR DESCRIPTION
**Pull request by Rohanmond**

### CODE REVIEW

File: src/contexts/data-context.js [hunk 1]
```diff
@@ -31,8 +40,86 @@ export const DataProvider = ({ children }) => {
...
+  };
+
+  const DeleteHistory = async ({ videoId }) => {
+    try {
+      const historyRes = await DeleteHistoryService({
+        videoId,
+        encodedToken: token,
+      });
+      if (historyRes.status === 200 || historyRes.status === 201) {
+        console.log(historyRes);
```
Comment (anuj-kumary) [inline src/contexts/data-context.js]: `Remove console.log`

**Review by anuj-kumary — APPROVED** [review]
Look good

I really liked:

- The UI, especially the chips components.
- How you break everything into small pieces.

**Review by PeeyushPrashant — APPROVED** [review]
folder structure is good
good naming conventions
looks good to me

**Review by mdarif6 — APPROVED** [review]
- I liked ellipsis  in videos thumbnail , want to know how it is working.
- Every thing looking good.

**Review by vinc3nati — APPROVED** [review]
Nice use of custom hooks and great component heirarchy

File: src/contexts/data-context.js (hunk 1 above)
Comment (Rohanmond) [reply src/contexts/data-context.js]: sure

**Pull request merged**